### PR TITLE
Update database interface to support get, create and update users

### PR DESCRIPTION
**Pull request by nathan96g**

### CODE REVIEW

File: app/src/main/java/ch/epfl/qedit/backend/database/FirebaseDBService.java [hunk 1]
```diff
@@ -70,4 +75,51 @@ private static void error(CompletableFuture<?> future, String message) {
...
+    public CompletableFuture<Void> createUser(String userId, String firstName, String lastName) {
+        Map<String, Object> data = new HashMap<>();
+        data.put("first_name", firstName);
+        data.put("state", lastName);
+        data.put("score", 0);
+        data.put("successes", 0);
+        data.put("attempts", 0);
+        data.put("quizzes", new HashMap<>());
+
+        // in firestore API it is specified that if there is no user with id userId, on will be
```
Comment (Anthony-Iozzia) [inline app/src/main/java/ch/epfl/qedit/backend/database/FirebaseDBService.java:102]: typo: "onE will be created"

File: app/src/main/java/ch/epfl/qedit/backend/database/FirebaseDBService.java [hunk 2]
```diff
@@ -70,4 +75,51 @@ private static void error(CompletableFuture<?> future, String message) {
...
+                .addOnFailureListener(e -> error(future, "The required user does not exist"));
+
+        return future;
+    }
+
+    @Override
+    public CompletableFuture<Void> createUser(String userId, String firstName, String lastName) {
+        Map<String, Object> data = new HashMap<>();
+        data.put("first_name", firstName);
+        data.put("state", lastName);
```
Comment (Anthony-Iozzia) [inline app/src/main/java/ch/epfl/qedit/backend/database/FirebaseDBService.java:96]: Hmmm, shouldn't it be data.put("last_name", lastName);?

File: app/src/test/java/ch/epfl/qedit/model/UserTest.java [hunk 3]
```diff
@@ -27,7 +27,7 @@ public void dummyTestUser() {
     }
 
     @Test
-    public void testConstructor() {
+    public void testConstructor1() {
```
Comment (nbaehler) [inline app/src/test/java/ch/epfl/qedit/model/UserTest.java:30]: There is only one of them but who cares ;)

Comment (nathan96g) [reply app/src/test/java/ch/epfl/qedit/model/UserTest.java:30]: oh yes right XD

**Pull request merged**